### PR TITLE
[SYCL] Add diagnostic test for global_variable_allowed attribute

### PR DESCRIPTION
**Pull request by schittir**
Add cases for this attribute independent of device_global

### CODE REVIEW

File: clang/test/SemaSYCL/global_variable_allowed.cpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
+// RUN: %clang_cc1 -fsycl-is-device -std=c++17 -sycl-std=2020 -verify -fsyntax-only %s
+
+#include "Inputs/sycl.hpp"
```
Comment (smanna12) [inline clang/test/SemaSYCL/global_variable_allowed.cpp:3]: Please add a comment about what test is doing. Thank you.

Comment (schittir) [reply clang/test/SemaSYCL/global_variable_allowed.cpp:3]: Done.

**Review by Fznamznon — COMMENTED** [review]
Please align the test with contributing guidelines https://github.com/intel/llvm/blob/sycl/sycl/doc/developer/ContributeToDPCPP.md#dpc-clang-fe-tests .

**Comment by schittir** [comment]
> Please align the test with contributing guidelines https://github.com/intel/llvm/blob/sycl/sycl/doc/developer/ContributeToDPCPP.md#dpc-clang-fe-tests .

Hi @Fznamznon could you please tell me which lines you would like changed?

File: clang/test/SemaSYCL/global_variable_allowed.cpp [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
```
Comment (Fznamznon) [inline clang/test/SemaSYCL/global_variable_allowed.cpp:3]:
```suggestion
#include "sycl.hpp"
```

File: clang/test/SemaSYCL/global_variable_allowed.cpp [hunk 3]
```diff
@@ -0,0 +1,65 @@
+// RUN: %clang_cc1 -fsycl-is-device -std=c++17 -sycl-std=2020 -verify -fsyntax-only %s
```
Comment (Fznamznon) [inline clang/test/SemaSYCL/global_variable_allowed.cpp:1]:
```suggestion
// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -std=c++17 -sycl-std=2020 -verify -fsyntax-only %s
```

File: clang/test/SemaSYCL/global_variable_allowed.cpp [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+// expected-error@+1{{'global_variable_allowed' attribute only applies to classes}}
+[[__sycl_detail__::global_variable_allowed]] int *pointer;
+
+// expected-error@+1{{'global_variable_allowed' attribute takes no arguments}}
+struct [[__sycl_detail__::global_variable_allowed(72)]] attribute_argument;
+
+
+int main() {
+  sycl::kernel_single_task<class KernelName1>([=]() {
```
Comment (Fznamznon) [inline clang/test/SemaSYCL/global_variable_allowed.cpp:54]:
```suggestion
sycl::queue q;
q.submit([&](sycl::handler &h) {
  h.single_task<class KernelName1>([=]() {
```

**Review by Fznamznon — APPROVED** [review]

**Review by smanna12 — APPROVED** [review]

**Pull request merged**